# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


In [1]:
import pandas as pd

### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [12]:
# Your code here
df = pd.read_csv("../DATA/car-sales.csv")
df.shape
df.head(10)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [9]:
# Your code here
df.describe(include="all")

Manufacturer Colour       Odometer      Doors         Price
count            10     10      10.000000  10.000000     10.000000
unique            4      5            NaN        NaN           NaN
top          Toyota  White            NaN        NaN           NaN
freq              4      4            NaN        NaN           NaN
mean            NaN    NaN   78601.400000   4.000000   7645.000000
std             NaN    NaN   61983.471735   0.471405   5379.407753
min             NaN    NaN   11179.000000   3.000000   3500.000000
25%             NaN    NaN   35836.250000   4.000000   4625.000000
50%             NaN    NaN   57369.000000   4.000000   6625.000000
75%             NaN    NaN   96384.500000   4.000000   7375.000000
max             NaN    NaN  213095.000000   5.000000  22000.000000

### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [11]:
# Your code here
car_count = df.Manufacturer.count()
car_count

10

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [15]:
# Your code here
df.groupby(['Manufacturer']).size()

Manufacturer
BMW       1
Honda     3
Nissan    2
Toyota    4
dtype: int64

### 5. Đếm số xe theo màu


In [17]:
# Your code here
df.groupby(["Colour"]).size()

Colour
Black    1
Blue     3
Green    1
Red      1
White    4
dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [20]:
# Your code here
df.Price.sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [21]:
# Your code here
top_price = df.nlargest(3, "Price")[["Manufacturer", "Price"]]
top_price

Manufacturer    Price
3          BMW  22000.0
9       Nissan   9700.0
6        Honda   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [22]:
# Your code here
top_odometer = df.nlargest(3, "Odometer")[["Manufacturer", "Odometer"]]
top_odometer

Manufacturer  Odometer
4       Nissan    213095
0       Toyota    150043
5       Toyota     99213

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [33]:
# Your code here
car_honda = df[df["Manufacturer"] == "Honda"]
car_honda_rate = len(car_honda) / len(df)
car_honda_price_mean = car_honda["Price"].mean()
car_honda_colour = car_honda["Colour"].unique()
print("Tỷ lệ xe Honda trong dữ liệu của cửa hàng là:", car_honda_rate)
print("Giá trung bình của xe hãng Honda của cửa hàng là:", car_honda_price_mean)
car_honda_colour



Tỷ lệ xe Honda trong dữ liệu của cửa hàng là: 0.3
Giá trung bình của xe hãng Honda của cửa hàng là: 6500.0


array(['Red', 'Blue'], dtype=object)

# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [35]:
# Your code here
price_mean = df.Price.mean()
price_medium = df.Price.median()
price_range = df.Price.max() - df.Price.min()
price_pvi = df.Price.quantile(0.75) - df.Price.quantile(0.25)
price_pvi_40 = df.Price.quantile(0.4)
price_pvi_60 = df.Price.quantile(0.6)

print(price_mean)
print(price_medium)
print(price_range)
print(price_pvi)
print(price_pvi_40)
print(price_pvi_60)


7645.0
6625.0
18500.0
2750.0
5750.0
7000.0


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [3]:
# Your code here
df = pd.read_csv("../DATA/tips.csv")
df.head(5)

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4

2. Xoá cột `Unnamed: 0`


In [5]:
# Your code here
df.drop(columns=["Unnamed: 0"])

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [6]:
# Your code here
df.shape

(244, 8)

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [13]:
# Your code here
df.describe(include='all')
df.dtypes

Unnamed: 0      int64
total_bill    float64
tip           float64
sex            object
smoker         object
day            object
time           object
size            int64
dtype: object

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [24]:
# Your code here
sex_count = df.sex.value_counts()
print(sex_count)
print("Tỷ lệ nam: ", sex_count["Male"]/len(df.sex))
print("Tỷ lệ nữ:", sex_count["Female"]/len(df.sex))

sex
Male      157
Female     87
Name: count, dtype: int64
Tỷ lệ nam:  0.6434426229508197
Tỷ lệ nữ: 1.0


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [27]:
# Your code here
smoke_count = df.smoker.value_counts()
smoke_count
print("Tỷ lê người hút thuốc:", smoke_count["Yes"]/len(df.smoker))

Tỷ lê người hút thuốc: 0.38114754098360654


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [34]:
# Your code here
day_counts = df.day.value_counts()
day_counts
print("Ngày trong tuần khách hàng hay đến dùng bữa nhất là", day_counts.idxmax())
print("Tỷ lệ", day_counts[day_counts.idxmax()] / len(df))

Ngày trong tuần khách hàng hay đến dùng bữa nhất là Sat
Tỷ lệ 0.35655737704918034


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [36]:
# Your code here
time_counts = df['time'].value_counts()
print("Nhà hàng phục vụ nhiều nhất là bữa ăn", time_counts.idxmax())

Nhà hàng phục vụ nhiều nhất là bữa ăn Dinner


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [38]:
# Your code here
size_max = df['size'].max()
size_max

6

10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [40]:
# Your code here
print(int(df['size'].mean()))

2


---